src/tools/payouts/query-payouts.ts

```diff
@@ -0,0 +1,103 @@
+import { z } from "zod";
+import { logger } from "../../server.js";
+import { accountPayoutQuerySchema } from "../../schemas/schemas.js";
+
+const QUERY_API_PATH = "/accountPayouts/events";
+
+export async function queryAccountPayouts(
+  params: z.infer<typeof accountPayoutQuerySchema>
+) {
+  const queryParams = new URLSearchParams();
+
+  if (params.startDate) queryParams.set("startDate", params.startDate);
+  if (params.endDate) queryParams.set("endDate", params.endDate);
+  if (params.pageSize) queryParams.set("pageSize", params.pageSize.toString());
+  if (params.narrative) queryParams.set("narrative", params.narrative);
+  if (params.transactionReference)
+    queryParams.set("transactionReference", params.transactionReference);
+  if (params.accountNumber)
+    queryParams.set("accountNumber", params.accountNumber);
+  if (params.payoutInstrumentId)
+    queryParams.set("payoutInstrumentId", params.payoutInstrumentId);
+  if (params.payoutInstrumentReference)
+    queryParams.set(
+      "payoutInstrumentReference",
+      params.payoutInstrumentReference
+    );
+  if (process.env.MERCHANT_ENTITY)
+    queryParams.set("entity", process.env.MERCHANT_ENTITY);
+  if (params.countryCode) queryParams.set("countryCode", params.countryCode);
+  if (params.pageNumber)
+    queryParams.set("pageNumber", params.pageNumber.toString());
+  if (params.paymentState) queryParams.set("paymentState", params.paymentState);
+  if (params.payeeName) queryParams.set("payeeName", params.payeeName);
+  if (params.sourceCurrency)
+    queryParams.set("sourceCurrency", params.sourceCurrency);
+  if (params.sourceAmount)
+    queryParams.set("sourceAmount", params.sourceAmount.toString());
+  if (params.targetCurrency)
+    queryParams.set("targetCurrency", params.targetCurrency);
+  if (params.targetAmount)
+    queryParams.set("targetAmount", params.targetAmount.toString());
+  return callQueryAPIWithParams(queryParams);
+}
+
+async function callQueryAPIWithParams(queryParams: URLSearchParams) {
+  try {
+    logger.info(`Calling GET ${QUERY_API_PATH}?${queryParams}`);
+
+    const response = await fetch(`${process.env.WORLDPAY_URL}${QUERY_API_PATH}?${queryParams}`, {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        "WP-Api-Version": "2025-01-01",
+        Authorization: `Basic ${Buffer.from(
+          `${process.env.WORLDPAY_USERNAME}:${process.env.WORLDPAY_PASSWORD}`
+        ).toString("base64")}`,
+      },
+    });
```

Comment:
```suggestion
    const base64Auth = Buffer.from(`${process.env.WORLDPAY_USERNAME}:${process.env.WORLDPAY_PASSWORD}`).toString("base64");
    const response = await fetch(`${process.env.WORLDPAY_URL}${QUERY_API_PATH}?${queryParams}`, {
      method: "GET",
      headers: {
        Accept: "application/json",
        "WP-Api-Version": "2025-01-01",
        Authorization: `Basic ${base64Auth}`,
      },
    });
```

This is more of a personal preference, but it would be better to extract the base64Auth into a variable and then use the string interpolation maybe we could even add some validation to guarantee that these values exist before running the http request 

Something like: 
```
const buildBasicAuthHeader = (user: string, pass: string): string => {
  if (!user || !pass) {
    throw new Error("Missing WORLDPAY_USERNAME or WORLDPAY_PASSWORD");
  }
  const token = Buffer.from(`${user}:${pass}`).toString("base64");
  return `Basic ${token}`;
}
```

Not sure if the Error could also help unmask issues by the agent, if we raise an error would it prompt the user for these?